### DISCUSSION TITLE
state.get returns content instead of raw value

### DISCUSSION THREAD

**Discussion by lumio** [typst/typst#3895] [General]
### Description

Hi everyone,

I'm currently trying to create a template that reads a list of items and generates content from it at the end of a document.

However, it seems that `state.get()` returns a content value instead of the documented `any` which I assumed means, the initial type.

### Reproduction

```typst
#let arr = (1, 2)
#arr.at(0)
#let arrState = state("arrState", ())
#arrState.update(a => a + (1,))
#arrState.update(a => a + (2,))
#let arrStateValue = context arrState.get() // context arrState.final() results in the same
#arrStateValue
#arrStateValue.at(0) // the error
```

This results in the following error:

```
error: expected string, found integer
line 8
```

### Expectation

Similar to line 2, I expected the same result at the last line.

If you need any more information, let me know :)

### Reproduction URL

https://typst.app/project/rccir0tQw_J7v2A1itRAqh

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
Because context returns opaque content, you need to do all your work on the value inside a context block.

See https://typst.app/docs/reference/context/

**Reply by lumio** [reply]
Got it. Thank you for your quick reply!
